src/core_codemods/literal_or_new_object_identity.py

```diff
@@ -0,0 +1,63 @@
+import libcst as cst
+from codemodder.codemods.api import BaseCodemod
+from codemodder.codemods.base_codemod import ReviewGuidance
+
+from codemodder.codemods.utils_mixin import NameResolutionMixin
+
+
+class LiteralOrNewObjectIdentity(BaseCodemod, NameResolutionMixin):
+    NAME = "literal-or-new-object-identity"
+    SUMMARY = "Replaces is operator with == for literal or new object comparisons"
+    REVIEW_GUIDANCE = ReviewGuidance.MERGE_WITHOUT_REVIEW
+    DESCRIPTION = SUMMARY
+    REFERENCES = [
+        {
+            "url": "https://docs.python.org/3/library/stdtypes.html#comparisons",
+            "description": "",
+        },
+    ]
+    CHANGE_DESCRIPTION = "Replaces is operator with =="
+
+    def _is_object_creation_or_literal(self, node: cst.BaseExpression):
+        match node:
+            case cst.List() | cst.Dict() | cst.Tuple() | cst.Set() | cst.Integer() | cst.Float() | cst.Imaginary() | cst.SimpleString() | cst.ConcatenatedString() | cst.FormattedString():
+                return True
+            case cst.Call(func=cst.Name() as name):
+                return self.is_builtin_function(node) and name.value in (
+                    "dict",
+                    "list",
+                    "tuple",
+                    "set",
+                )
+        return False
+
+    def leave_Comparison(
+        self, original_node: cst.Comparison, updated_node: cst.Comparison
+    ) -> cst.BaseExpression:
+        if self.filter_by_path_includes_or_excludes(self.node_position(original_node)):
```

Comment:
I realized I've forgotten this filter in many of my own recent codemods so I'll need to go back and make updates.

Reply:
Another reason to have another look at the API. This should be automated in these simple cases.